
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="../5_reconciliation/">
      
      
        <link rel="next" href="../../day3/1_general/">
      
      
      <link rel="icon" href="../../../assets/images/general/SIB_logo.svg">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.14">
    
    
      
        <title>Gene Family Ancestral State Reconstruction - Biodiversity bioinformatics</title>
      
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/main.342714a4.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../../stylesheets/extra.css">
    
    <script>__md_scope=new URL("../../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    <body dir="ltr">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#aim" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <div data-md-color-scheme="default" data-md-component="outdated" hidden>
        
          <aside class="md-banner md-banner--warning">
            <div class="md-banner__inner md-grid md-typeset">
              
  You're not viewing the latest version.
  <a href="../../../.."> 
    <strong>Click here to go to latest.</strong>
  </a>

            </div>
            <script>var el=document.querySelector("[data-md-component=outdated]"),base=new URL("../../.."),outdated=__md_get("__outdated",sessionStorage,base);!0===outdated&&el&&(el.hidden=!1)</script>
          </aside>
        
      </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../../.." title="Biodiversity bioinformatics" class="md-header__button md-logo" aria-label="Biodiversity bioinformatics" data-md-component="logo">
      
  <img src="../../../assets/images/general/SIB_logo.svg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Biodiversity bioinformatics
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Gene Family Ancestral State Reconstruction
            
          </span>
        </div>
      </div>
    </div>
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/sib-swiss/biodiversity-bioinformatics" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8M97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg>
  </div>
  <div class="md-source__repository">
    sib-swiss/biodiversity-bioinformatics
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../../.." title="Biodiversity bioinformatics" class="md-nav__button md-logo" aria-label="Biodiversity bioinformatics" data-md-component="logo">
      
  <img src="../../../assets/images/general/SIB_logo.svg" alt="logo">

    </a>
    Biodiversity bioinformatics
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/sib-swiss/biodiversity-bioinformatics" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8M97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg>
  </div>
  <div class="md-source__repository">
    sib-swiss/biodiversity-bioinformatics
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../.." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Home
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../precourse/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Precourse preparations and Gitpod information
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../course_schedule/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Course schedule
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" >
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Day 1 - Introduction to molecular evolution & phylogenetics
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            Day 1 - Introduction to molecular evolution & phylogenetics
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day1/1_general/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    General information and course material
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day1/2_exercices/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Links to the OMA Academy modules
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" checked>
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Day 2 - Orthology-focused arthropod comparative genomics
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            Day 2 - Orthology-focused arthropod comparative genomics
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../1_general/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    General information and course material
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../2_software/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Software and resources
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../3_busco/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    BUSCO - the what, why, and how!
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../4_homology/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Homology, Orthology, & Paralogy
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../5_reconciliation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Gene-Tree-Species-Tree Reconciliation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    Gene Family Ancestral State Reconstruction
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    Gene Family Ancestral State Reconstruction
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#aim" class="md-nav__link">
    <span class="md-ellipsis">
      Aim
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#learning-outcomes" class="md-nav__link">
    <span class="md-ellipsis">
      Learning outcomes
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#using-orthologue-counts-to-estimate-ancestral-gene-content" class="md-nav__link">
    <span class="md-ellipsis">
      Using orthologue counts to estimate ancestral gene content
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#comparing-ancestral-gene-content-estimates-with-reconciliation-results" class="md-nav__link">
    <span class="md-ellipsis">
      Comparing ancestral gene content estimates with reconciliation results
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#congratulations-you-made-it-to-the-end" class="md-nav__link">
    <span class="md-ellipsis">
      Congratulations, you made it to the end!
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6" >
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Day 3 - Duplication of genes and genomes, expression
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            Day 3 - Duplication of genes and genomes, expression
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day3/1_general/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    General information and course material
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day3/2_software/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Software and R packages
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day3/3_gene_function/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Finding information on gene function at small scale
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day3/4_group_function/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Finding information about function of a group of genes
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day3/5_duplicated_function/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Function of duplicated genes
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#aim" class="md-nav__link">
    <span class="md-ellipsis">
      Aim
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#learning-outcomes" class="md-nav__link">
    <span class="md-ellipsis">
      Learning outcomes
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#using-orthologue-counts-to-estimate-ancestral-gene-content" class="md-nav__link">
    <span class="md-ellipsis">
      Using orthologue counts to estimate ancestral gene content
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#comparing-ancestral-gene-content-estimates-with-reconciliation-results" class="md-nav__link">
    <span class="md-ellipsis">
      Comparing ancestral gene content estimates with reconciliation results
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#congratulations-you-made-it-to-the-end" class="md-nav__link">
    <span class="md-ellipsis">
      Congratulations, you made it to the end!
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  
  


<h2 id="aim">Aim</h2>
<p>Estimate ancestral gene copy numbers given the species phylogeny and counts of orthologous genes present in extant species.</p>
<h2 id="learning-outcomes">Learning outcomes</h2>
<ul>
<li>What is the difference between tree reconciliation and state reconstruction?</li>
<li>What is gene family ancestral state reconstruction trying to achieve and how?</li>
</ul>
<h2 id="using-orthologue-counts-to-estimate-ancestral-gene-content">Using orthologue counts to estimate ancestral gene content</h2>
<p><img alt="" src="../../../assets/images/day2/session4/cafe_logo.jpg" style="float: right;" width="250" /></p>
<p>There are a huge range of tools designed for ancestral state reconstructions, you may for example have encountered the R package, <a href="http://www.phytools.org/">phytools</a>. Today we will use <a href="https://github.com/hahnlab/CAFE5">CAFE5</a>, a tool specifically designed for working with gene count data to infer ancestral gene contents per gene family.</p>
<p>Before we reconstruct the ancestral state gene content of our species of interest, let&rsquo;s create a working directory for this exercise, starting by opening a terminal on the Workspace if you&rsquo;ve not already got one open. Then, from the <code>/workspace/biodivinfo/</code> directory, create a new directory (<code>mkdir</code>) and navigate into the new directory (<code>cd</code>):</p>
<div class="highlight"><pre><span></span><code>cd /workspace/biodivinfo/
mkdir Session4
cd Session4/
</code></pre></div>
<p>We will continue to work with the same dataset of orthologous groups delineated across 15 beetles from OrthoDB. The main input requirements for CAFE are:</p>
<ul>
<li>An ultrametric time-calibrated species tree<ul>
<li>We have the <strong>ultrametric time-calibrated species tree</strong> we made earlier: <code>Coleoptera_TimeTree_IDs.tre</code></li>
<li>This is a version with just the species identifiers not the names. You can fetch it from the data folder:
<div class="highlight"><pre><span></span><code>cp /workspace/biodivinfo/data/Session4/Coleoptera_TimeTree_IDs.tre .
</code></pre></div></li>
</ul>
</li>
<li>A table of counts of orthologues per species<ul>
<li>This table was obtained by downloading and parsing the full table of orthology data from OrthoDB to extract counts of genes per species for <em>Coleoptera</em>-level orthogroups, you can fetch these pre-processed data here:
<div class="highlight"><pre><span></span><code>cp /workspace/biodivinfo/data/Session4/Coleoptera_OGs.tsv .
</code></pre></div></li>
<li>
<p>The table contains rows of orthogroups (blue) with columns of species (green) and all the counts of orthologues per orthogroup per species</p>
<ul>
<li>For the class the full set of <strong>14&lsquo;753</strong> orthogroups has been first filtered to select only orthologous groups found in more than 90% of species (with no restriction on gene copy-number), leaving <strong>6&lsquo;824</strong> orthogroups. Note that the <strong>species names/identifiers</strong> must be the same as those we have in our <strong>ultrametric time-calibrated species phylogeny</strong></li>
</ul>
<p><figure>
  <img src="../../../assets/images/day2/session4/orthologues_counts.jpg" align="center" width=600/>
</figure></p>
</li>
</ul>
</li>
</ul>
<p>Start by checking out what <code>CAFE</code> needs as command-line options, input, and output names etc. in order to perform an ancestral state reconstruction:</p>
<div class="highlight"><pre><span></span><code>cafe5 -h
</code></pre></div>
<figure>
  <img src="../../../assets/images/day2/session4/cafe_usage.jpg" align="center" width=600/>
</figure>

<p>The main input that CAFE needs:</p>
<ul>
<li><code>-i</code>, <code>--infile</code>: path to tab delimited gene families file to be analysed</li>
<li><code>-t</code>, <code>--tree</code>: path to file containing Newick formatted tree</li>
</ul>
<p>Putting this all together we have the following:
<div class="highlight"><pre><span></span><code>cafe5 -i Coleoptera_OGs.tsv -t Coleoptera_TimeTree_IDs.tre
</code></pre></div></p>
<figure>
  <img src="../../../assets/images/day2/session4/cafe_run.jpg" align="center" width=600/>
</figure>

<p>CAFE uses a <strong>birth and death process</strong> to model gene gain and loss across a user-specified phylogenetic tree. The distribution of family sizes generated under this model provides a basis for assessing the <strong>significance of the observed family size differences</strong> among taxa. Estimating Lambda (the traditional running of CAFE) performs a maximum likelihood calculation to estimate the most likely rate of change across the entire tree, given all the families and their counts</p>
<p>For a typical CAFE analysis, users are most interested in determining two things:</p>
<ul>
<li>Which gene families/orthogroups are rapidly evolving?</li>
<li>On which branches of the tree these families are rapidly evolving?</li>
</ul>
<p>By default, CAFE creates its outputs into a folder called &ldquo;results&rdquo;:
<div class="highlight"><pre><span></span><code>ls -l /workspace/biodivinfo/Session4/results
</code></pre></div></p>
<figure>
  <img src="../../../assets/images/day2/session4/cafe_results1.jpg" align="center" width=250/>
</figure>

<div class="admonition note">
<p class="admonition-title">If CAFE failed, you can get the file from the folder with data files for the practical instead</p>
<div class="highlight"><pre><span></span><code>cp -r /workspace/biodivinfo/data/Session4/results .
</code></pre></div>
</div>
<p>The <code>Base_asr.tre</code> output file contains all orthogroups with the ancestral gene counts mapped to the species tree. You might recognise the first tree as the orthogroup we used to build our first species tree (<code>10000at7041)</code>.</p>
<figure>
  <img src="../../../assets/images/day2/session4/cafe_results2.jpg" align="center" width=600/>
</figure>

<p>We can get an overview of gains and losses across all families using a script provided by CAFE, <code>cafe5_draw_tree.py</code>:
<div class="highlight"><pre><span></span><code>wget https://raw.githubusercontent.com/hahnlab/CAFE5/master/docs/tutorial/cafe5_draw_tree.py
</code></pre></div></p>
<div class="admonition note">
<p class="admonition-title">You can also fetch the python script from the folder with data files for the practical</p>
<div class="highlight"><pre><span></span><code>cp /workspace/biodivinfo/scripts/cafe5_draw_tree.py .
</code></pre></div>
</div>
<figure>
  <img src="../../../assets/images/day2/session4/cafe_git.jpg" align="center" width=600/>
</figure>

<p>Start by checking out what are the script parameters:
<div class="highlight"><pre><span></span><code>python cafe5_draw_tree.py -h
</code></pre></div></p>
<figure>
  <img src="../../../assets/images/day2/session4/cafe_script.jpg" align="center" width=600/>
</figure>

<p>First we can plot the gains (Increase) across the tree:
<div class="highlight"><pre><span></span><code>python cafe5_draw_tree.py -i results/Base_clade_results.txt -d results/Base_report.cafe -y Increase -o CAFE_gains_tree.png
</code></pre></div></p>
<p>And now we can plot the losses (Decrease) across the tree:
<div class="highlight"><pre><span></span><code>python cafe5_draw_tree.py -i results/Base_clade_results.txt -d results/Base_report.cafe -y Decrease -o CAFE_losses_tree.png
</code></pre></div></p>
<figure>
  <img src="../../../assets/images/day2/session4/cafe_script_run.jpg" align="center" width=600/>
</figure>

<p>The <code>.png</code> output files should be in your working directory, you can open them in Gitpod to view numbers of families showing gains or losses across the species tree.</p>
<figure>
  <img src="../../../assets/images/day2/session4/cafe_trees.jpg" align="center" width=600/>
</figure>

<div class="admonition note">
<p class="admonition-title">High-resolution version</p>
<p>You can find a better version of the figure on Gitpod at <code>/workspace/biodivinfo/data/Session4/CAFE_gains_tree.png</code> and <code>/workspace/biodivinfo/data/Session4/CAFE_losses_tree.png</code>. Or you can also see them here: <a href="https://gitlab.com/evogenlab/teaching-repos/biodivinfo/-/blob/main/data/Session4/CAFE_gains_tree.png">gains</a> &amp; <a href="https://gitlab.com/evogenlab/teaching-repos/biodivinfo/-/blob/main/data/Session4/CAFE_losses_tree.png">losses</a></p>
</div>
<p><strong>Questions:</strong></p>
<ul>
<li><strong>Which internal node shows the most gains?</strong></li>
<li><strong>Which internal node shows the most losses?</strong></li>
</ul>
<details class="done">
<summary>Answer</summary>
<ul>
<li>The node between species <code>224129</code> and <code>7054</code> shows the most gains (<strong>230</strong>).</li>
<li>The node between species <code>115357</code> and <code>7070</code> shows the most losses (<strong>116</strong>).</li>
</ul>
</details>
<p>A reminder of the species in our tree:</p>
<figure>
  <img src="../../../assets/images/day2/session4/species_tree.jpg" align="center" width=600/>
</figure>

<h2 id="comparing-ancestral-gene-content-estimates-with-reconciliation-results">Comparing ancestral gene content estimates with reconciliation results</h2>
<p>As a final step we can compare the results from our previous gene-tree-species-tree reconciliation of orthogroup <code>10261at7041</code> with those from gene family ancestral state reconstruction.</p>
<p>Extract the gain-loss annotated tree for <code>10261at7041</code> from the CAFE results:
<div class="highlight"><pre><span></span><code>grep 10261at7041 results/Base_asr.tre
</code></pre></div></p>
<figure>
  <img src="../../../assets/images/day2/session4/grep_tree.jpg" align="center" width=600/>
</figure>

<p>Copy the output of the previous <code>grep</code> command (which is a Newick tree) and open it in iTOL as we did before - note that the Newick tree starts with the first open parenthesis and ends with the semicolon. Use the <code>Advanced Control Panel</code> to Display the Node IDs and view internal node annotations.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>You can also find the Newick file on gitpod at: <code>/workspace/biodivinfo/data/Session4/10261at7041_reconstruction.nwk</code></p>
</div>
<figure>
  <img src="../../../assets/images/day2/session4/itol_settings.jpg" align="center" width=250/>
</figure>

<div class="admonition note">
<p class="admonition-title">If you did not manage to visualise the tree</p>
<p>You can find it on Gitpod at <code>/workspace/biodivinfo/data/Session4/tree_10261at7041_CAFE.jpg</code>. Or you can directly see it <a href="https://gitlab.com/evogenlab/teaching-repos/biodivinfo/-/blob/main/data/Session4/tree_10261at7041_CAFE.jpg">here</a>.</p>
</div>
<p><img alt="" src="../../../assets/images/day2/session4/itol_labels.jpg" style="float: right;" width="300" /></p>
<p>Terminal nodes (leaves) are labelled with the species (green), CAFE&rsquo;s own numbering of all nodes in the tree (yellow) and finally gene the copy number (pink). Nodes inside the tree are labelled with the internal CAFE numbering of all nodes in the tree (yellow) and the gene the copy number (pink).</p>
<p>Nodes exhibiting significant changes from their ancestral states are indicated with an asterisk (*). We can now place our CAFE estimates next to our treerecs estimates to compare the results from the two approaches:</p>
<figure>
  <img src="../../../assets/images/day2/session4/itol_tree.jpg" align="center"/>
</figure>

<p><strong>Question: Do the ancestral state reconstructions results (<code>CAFE</code>) agree with the tree reconciliation events (<code>treerecs</code> - after asking it to ignore low-confidence nodes)?</strong></p>
<details class="done">
<summary>Answer</summary>
<p>Yes, they perfectly agree with each other!</p>
</details>
<p><br></p>
<h1 id="congratulations-you-made-it-to-the-end">Congratulations, you made it to the end!</h1>
<figure>
  <img src="../../../assets/images/day2/session4/end.jpg" align="center"/>
</figure>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright &copy; 2021-2025 SIB Swiss institute of Bioinformatics - <a href="#__consent">Change cookie settings</a>
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "../../..", "features": ["announce.dismiss", "content.code.copy"], "search": "../../../assets/javascripts/workers/search.d50fe291.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": {"provider": "mike"}}</script>
    
    
      <script src="../../../assets/javascripts/bundle.13a4f30d.min.js"></script>
      
    
  </body>
</html>