
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="../4_homology/">
      
      
        <link rel="next" href="../6_reconstruction/">
      
      
      <link rel="icon" href="../../../assets/images/general/SIB_logo.svg">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.14">
    
    
      
        <title>Gene-Tree-Species-Tree Reconciliation - Biodiversity bioinformatics</title>
      
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/main.342714a4.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../../stylesheets/extra.css">
    
    <script>__md_scope=new URL("../../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    <body dir="ltr">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#aim" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <div data-md-color-scheme="default" data-md-component="outdated" hidden>
        
          <aside class="md-banner md-banner--warning">
            <div class="md-banner__inner md-grid md-typeset">
              
  You're not viewing the latest version.
  <a href="../../../.."> 
    <strong>Click here to go to latest.</strong>
  </a>

            </div>
            <script>var el=document.querySelector("[data-md-component=outdated]"),base=new URL("../../.."),outdated=__md_get("__outdated",sessionStorage,base);!0===outdated&&el&&(el.hidden=!1)</script>
          </aside>
        
      </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../../.." title="Biodiversity bioinformatics" class="md-header__button md-logo" aria-label="Biodiversity bioinformatics" data-md-component="logo">
      
  <img src="../../../assets/images/general/SIB_logo.svg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Biodiversity bioinformatics
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Gene-Tree-Species-Tree Reconciliation
            
          </span>
        </div>
      </div>
    </div>
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/sib-swiss/biodiversity-bioinformatics" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8M97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg>
  </div>
  <div class="md-source__repository">
    sib-swiss/biodiversity-bioinformatics
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../../.." title="Biodiversity bioinformatics" class="md-nav__button md-logo" aria-label="Biodiversity bioinformatics" data-md-component="logo">
      
  <img src="../../../assets/images/general/SIB_logo.svg" alt="logo">

    </a>
    Biodiversity bioinformatics
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/sib-swiss/biodiversity-bioinformatics" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8M97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg>
  </div>
  <div class="md-source__repository">
    sib-swiss/biodiversity-bioinformatics
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../.." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Home
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../precourse/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Precourse preparations and Gitpod information
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../course_schedule/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Course schedule
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" >
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Day 1 - Introduction to molecular evolution & phylogenetics
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            Day 1 - Introduction to molecular evolution & phylogenetics
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day1/1_general/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    General information and course material
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day1/2_exercices/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Links to the OMA Academy modules
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" checked>
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Day 2 - Orthology-focused arthropod comparative genomics
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            Day 2 - Orthology-focused arthropod comparative genomics
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../1_general/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    General information and course material
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../2_software/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Software and resources
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../3_busco/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    BUSCO - the what, why, and how!
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../4_homology/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Homology, Orthology, & Paralogy
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    Gene-Tree-Species-Tree Reconciliation
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    Gene-Tree-Species-Tree Reconciliation
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#aim" class="md-nav__link">
    <span class="md-ellipsis">
      Aim
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#learning-outcomes" class="md-nav__link">
    <span class="md-ellipsis">
      Learning outcomes
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#building-an-ultrametric-time-calibrated-species-tree" class="md-nav__link">
    <span class="md-ellipsis">
      Building an ultrametric time-calibrated species tree
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#building-a-tree-for-a-dynamic-gene-family" class="md-nav__link">
    <span class="md-ellipsis">
      Building a tree for a dynamic gene family
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#reconciling-the-gene-and-species-trees" class="md-nav__link">
    <span class="md-ellipsis">
      Reconciling the gene and species trees
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../6_reconstruction/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Gene Family Ancestral State Reconstruction
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6" >
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Day 3 - Duplication of genes and genomes, expression
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            Day 3 - Duplication of genes and genomes, expression
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day3/1_general/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    General information and course material
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day3/2_software/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Software and R packages
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day3/3_gene_function/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Finding information on gene function at small scale
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day3/4_group_function/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Finding information about function of a group of genes
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day3/5_duplicated_function/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Function of duplicated genes
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#aim" class="md-nav__link">
    <span class="md-ellipsis">
      Aim
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#learning-outcomes" class="md-nav__link">
    <span class="md-ellipsis">
      Learning outcomes
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#building-an-ultrametric-time-calibrated-species-tree" class="md-nav__link">
    <span class="md-ellipsis">
      Building an ultrametric time-calibrated species tree
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#building-a-tree-for-a-dynamic-gene-family" class="md-nav__link">
    <span class="md-ellipsis">
      Building a tree for a dynamic gene family
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#reconciling-the-gene-and-species-trees" class="md-nav__link">
    <span class="md-ellipsis">
      Reconciling the gene and species trees
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  
  


  <h1>Gene-Tree-Species-Tree Reconciliation</h1>

<h2 id="aim">Aim</h2>
<p>Convert the molecular species phylogeny into an ultrametric time-calibrated species tree and use this for gene-tree-species-tree reconciliation.</p>
<h2 id="learning-outcomes">Learning outcomes</h2>
<ul>
<li>What is the difference between a molecular and a time-calibrated phylogeny?</li>
<li>What are gene-tree-species-tree reconciliation algorithms trying to achieve and how?</li>
</ul>
<h2 id="building-an-ultrametric-time-calibrated-species-tree">Building an ultrametric time-calibrated species tree</h2>
<p><img alt="" src="../../../assets/images/day2/session3/ape_logo.jpg" style="float: right;" width="250" /></p>
<p>There are several tools available for converting molecular phylogenies to ultrametric time-calibrated trees. They usually require as input your molecular phylogeny as well as calibration points (estimates usually from fossils about the ranges of likely divergence times) for one or more nodes on the tree. We will use the <strong>chronos</strong> function of the <a href="http://ape-package.ird.fr/">ape</a> package in R.</p>
<p>Before we explore the tree, let&rsquo;s create a working directory for this exercise, starting by opening a terminal on the Workspace if you&rsquo;ve not already got one open. Then, from the <code>/workspace/biodivinfo/</code> directory, create a new directory (<code>mkdir</code>) and navigate into the new directory (<code>cd</code>):</p>
<div class="highlight"><pre><span></span><code>cd /workspace/biodivinfo/
mkdir Session3
cd Session3/
</code></pre></div>
<p>Using the same set of 15 beetle species, but now applying the concatenation approach to be able to use alignments from 1&lsquo;000 single-copy orthologous groups:</p>
<ul>
<li>Aligned each orthogroup</li>
<li>Filtered/trimmed each orthogroup</li>
<li>Concatenated the trimmed alignments into a superalignment</li>
<li>Used RAxML to compute, with bootstraps, the molecular species phylogeny</li>
</ul>
<p>This took several hours to run, therefore we are not going to try this in the class. Instead, the results are provided for you <a href="https://gitlab.com/evogenlab/teaching-repos/biodivinfo/-/blob/main/data/Session3/Step1/OGs-1000-moltree-names-rooted.tre">here</a> and also pasted below, where the tree has been rooted based on the known outgroup species (<code>7054_0</code> &amp; <code>224129_0</code>) and where internal labels and species names have been added to make it easier to interpret and compare with the current understanding of beetle phylogeny.</p>
<p>Here is the molecular species tree in Newick format:
<div class="highlight"><pre><span></span><code>((((((((50390_0_Diabrotica_virgifera:0.15666,7539_0_Leptinotarsa_decemlineata:0.15804)100:0.05974,217634_0_Anoplophora_glabripennis:0.11339)Chrysomeloidea[100]:0.04091,(7048_0_Sitophilus_oryzae:0.12494,77166_0_Dendroctonus_ponderosae:0.13601)Curculionoidea[100]:0.13858)100:0.02727,116153_0_Aethina_tumida:0.24260)100:0.03487,(115357_0_Harmonia_axyridis:0.06618,41139_0_Coccinella_septempunctata:0.07128)Coccinelloidea[100]:0.17882)88:0.01976,(41895_0_Tribolium_madens:0.02321,7070_0_Tribolium_castaneum:0.02060)Tenebrionoidea[100]:0.28640)100:0.09637,(110193_0_Nicrophorus_vespilloides:0.33251,(1629725_0_Oryctes_borbonicus:0.22252,166361_0_Onthophagus_taurus:0.24064)Scarabaeoidea[100]:0.07437)100:0.05341):0.00483,(224129_0_Agrilus_planipennis:0.31846,7054_0_Photinus_pyralis:0.29250):0.09174)100;
</code></pre></div></p>
<div class="admonition note">
<p class="admonition-title">Newick format - <a href="https://en.wikipedia.org/wiki/Newick_format">Wikipedia</a></p>
<ul>
<li>Parentheses indicate hierarchical relationships amongst species</li>
<li>Branches can be labelled with bootstrap values, <em>e.g.</em> 100</li>
<li>Branches can also be labelled with names, <em>e.g.</em> <em>Tenebrionoidea</em></li>
<li>Notice the semicolon at the end of the tree - it is often forgotten but most tools won&rsquo;t recognize a file as Newick format if it&rsquo;s missing!</li>
</ul>
</div>
<p>You can also copy the file from the provided data folder:
<div class="highlight"><pre><span></span><code>cp /workspace/biodivinfo/data/Session3/Step1/OGs-1000-moltree-names-rooted.tre .
</code></pre></div></p>
<p>Now to visualise the tree we will, as before, copy and paste this molecular species phylogeny to upload it to <a href="https://itol.embl.de/">iTOL</a>.</p>
<figure>
  <img src="../../../assets/images/day2/session3/itol_form.jpg" align="center" width=600/>
</figure>

<p>Use the iTOL <code>Advanced Control Panel</code> to turn on <code>Display</code> of <code>Node IDs</code> and turn on <code>Display</code> as <code>Text</code> the <code>Bootstraps</code>.</p>
<figure>
  <img src="../../../assets/images/day2/session3/itol_settings1.jpg" align="center" width=250/>
</figure>

<p>Notice that now there is only one node that did not converge to 100% bootstrap support - regarding the placement of the groups of <strong><em>Tenebrionoidea</em></strong> &amp; <strong><em>Coccinelloidea</em></strong>.</p>
<figure>
  <img src="../../../assets/images/day2/session3/itol_tree1.jpg" align="center" width=600/>
</figure>

<div class="admonition note">
<p class="admonition-title">If you did not manage to visualise the tree</p>
<p>You can find it on Gitpod at <code>/workspace/biodivinfo/data/Session3/Step1/tree_1000_OGs.jpg</code>. Or you can directly see it <a href="https://gitlab.com/evogenlab/teaching-repos/biodivinfo/-/blob/main/data/Session3/Step1/tree_1000_OGs.jpg">here</a>.</p>
</div>
<p>We now need to compare our tree to the literature to see if it is in agreement with the current thinking regarding the relations amongst beetle lineages.</p>
<p><strong>For comparison with the latest beetle phylogeny from the literature (see below):</strong></p>
<figure>
  <img src="../../../assets/images/day2/session3/itol_tree2.jpg" align="center" width=600/>
</figure>

<p>You should be able to find the internal node labels, provided for you on the tree you have just uploaded on iToL:</p>
<figure>
  <img src="../../../assets/images/day2/session3/beetle_clades.jpg" align="center" width=250/>
</figure>

<p><strong>Question: Does our 1&lsquo;000-orthogroup molecular species phylogeny agree with the current understanding of species relationships (see phylogeny below)?</strong></p>
<details class="done">
<summary>Answer</summary>
<p>Yes! Remember you can invert two branches of a node to facilitate the comparison between the two trees!</p>
</details>
<figure>
  <img src="../../../assets/images/day2/session3/beetle_tree.jpg" align="center" width=600/>
  <figcaption>Recent reference on Coleoptera phylogeny: Integrated phylogenomics and fossil data illuminate the evolution of beetles, Cai et al. 2022 </figcaption>
</figure>

<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Link to <a href="https://doi.org/10.1098/rsos.211771">Cai et al. 2022</a>.</p>
</div>
<p>We can now convert our molecular phylogeny to an ultrametric time-calibrated tree using the <code>chronos</code> function of the <code>ape</code> package.</p>
<div class="admonition note">
<p class="admonition-title"><a href="https://rdrr.io/cran/ape/man/chronos.html">Chronos</a>: Molecular Dating by Penalised Likelihood and Maximum Likelihood</p>
<ul>
<li><code>chronos</code> is the main function fitting a chronogram to a phylogenetic tree whose branch lengths are in number of substitution per sites</li>
<li><code>makeChronosCalib</code> is a tool to prepare data frames with the calibration points of the phylogenetic tree</li>
</ul>
</div>
<p>We could provide multiple calibration points to smooth the tree, but for today we will just find an estimate for the root age and use that as the only calibration point. We can use <a href="http://www.timetree.org/">TimeTree.org</a> to find a divergence estimate for any two beetle species in our collection that span the root of our tree, <em>e.g.</em> the age of the last common ancestor of <em>Agrilus planipennis</em> and <em>Leptinotarsa decemlineata</em>.</p>
<figure>
  <img src="../../../assets/images/day2/session3/timetree_home.jpg" align="center" width=600/>
</figure>

<p>TimeTree provides us with an &ldquo;Adjusted Time&rdquo; for the pairwise divergence time for <em>Agrilus Planipennis</em> and <em>Leptinotarsa Decemlineata</em> of 256 Million Years Ago (MYA).</p>
<figure>
  <img src="../../../assets/images/day2/session3/timetree_results.jpg" align="center" width=600/>
</figure>

<p><br>The R code for running chronos is provided here:
<div class="highlight"><pre><span></span><code><span class="c1"># Converting a molecular species phylogeny into a time-calibrated species tree</span>
<span class="c1"># If you do not have &#39;ape&#39; you will need to install it: install.packages(&quot;ape&quot;)</span>

<span class="c1"># Load ape package</span>
<span class="nf">library</span><span class="p">(</span><span class="n">ape</span><span class="p">)</span>

<span class="c1"># Read in the molecular phylogeny</span>
<span class="c1"># Make sure you are working in the same directory as the tree file</span>
<span class="n">moltree</span><span class="o">&lt;-</span><span class="nf">read.tree</span><span class="p">(</span><span class="s">&quot;OGs-1000-moltree-names-rooted.tre&quot;</span><span class="p">)</span>

<span class="c1"># Provide root calibration age of 256 million years</span>
<span class="n">calib</span><span class="o">&lt;-</span><span class="nf">makeChronosCalib</span><span class="p">(</span><span class="n">moltree</span><span class="p">,</span><span class="w"> </span><span class="n">node</span><span class="o">=</span><span class="s">&quot;root&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">age.min</span><span class="o">=</span><span class="m">256</span><span class="p">)</span>

<span class="c1"># use the chronos function to convert the tree with the age constraint</span>
<span class="n">timtree</span><span class="o">&lt;-</span><span class="nf">chronos</span><span class="p">(</span><span class="n">moltree</span><span class="p">,</span><span class="w"> </span><span class="n">calibration</span><span class="o">=</span><span class="n">calib</span><span class="p">,</span><span class="w"> </span><span class="n">lambda</span><span class="o">=</span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">model</span><span class="o">=</span><span class="s">&quot;discrete&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">control</span><span class="o">=</span><span class="nf">chronos.control</span><span class="p">(</span><span class="n">nb.rate.cat</span><span class="o">=</span><span class="m">1</span><span class="p">))</span>

<span class="c1"># check that the result is indeed ultrametric</span>
<span class="nf">is.ultrametric</span><span class="p">(</span><span class="n">timtree</span><span class="p">)</span>
<span class="nf">write.tree</span><span class="p">(</span><span class="n">timtree</span><span class="p">,</span><span class="w"> </span><span class="n">file</span><span class="o">=</span><span class="s">&quot;Coleoptera_TimeTree.tre&quot;</span><span class="p">)</span>
</code></pre></div></p>
<p>Or you can fetch the R script with:
<div class="highlight"><pre><span></span><code>cp /workspace/biodivinfo/scripts/MolecularTree_to_TimeTree.R .
</code></pre></div></p>
<div class="tabbed-set tabbed-alternate" data-tabs="1:2"><input checked="checked" id="__tabbed_1_1" name="__tabbed_1" type="radio" /><input id="__tabbed_1_2" name="__tabbed_1" type="radio" /><div class="tabbed-labels"><label for="__tabbed_1_1">If you want to run the script automatically</label><label for="__tabbed_1_2">If you want to run the successive steps manually</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<div class="highlight"><pre><span></span><code>Rscript MolecularTree_to_TimeTree.R
</code></pre></div>
</div>
<div class="tabbed-block">
<ul>
<li>Launch R: <code>R</code></li>
<li>Now for actually running chronos… first you need to load the library <code>ape</code>: <code>library("ape")</code></li>
<li>Then you can copy the R code provided above or from the provided R script file (<code>MolecularTree_to_TimeTree.R</code>) and enter return to execute the code<ul>
<li><code>read.tree</code>: read in your molecular phylogeny file</li>
<li><code>makeChronosCalib</code>: set the minimum age of the root of your tree</li>
<li><code>chronos</code>: smooth conversion of the &lsquo;moltree&rsquo; to the &lsquo;timetree&rsquo;</li>
<li><code>write.tree</code>: write the resulting timetree to a file (<code>Coleoptera_TimeTree.tre</code>)</li>
</ul>
</li>
<li>To exit R simply type <code>q()</code> and then return</li>
<li>No need to save your workspace image: <code>Save workspace image? [y/n/c]: n</code></li>
</ul>
</div>
</div>
</div>
<figure>
  <img src="../../../assets/images/day2/session3/ape_run.jpg" align="center" width=600/>
</figure>

<p>Now let&rsquo;s open the tree file produced by chronos (<code>Coleoptera_TimeTree.tre</code>) and use <a href="https://itol.embl.de/">iTOL</a> to view it (as before), you can find the file in the EXPLORER menu under Session3, copy the Newick formatted tree and go to <a href="https://itol.embl.de/">iTOL</a> to upload it.</p>
<figure>
  <img src="../../../assets/images/day2/session3/ape_timetree.jpg" align="center" width=600/>
</figure>

<p>The default view looks something like this, so we can use the <code>Advanced Control Panel</code> to change the views and thereby make comparisons with the literature tree easier.</p>
<figure>
  <img src="../../../assets/images/day2/session3/itol_tree3.jpg" align="center" width=600/>
</figure>

<p><img _="," alt="" src="../../../assets/images/day2/session3/itol_settings2.jpg" style="float: right;" width="300" /></p>
<p>Use the <code>Advanced Control Panel</code> to display the tree scale in inverted direction:</p>
<ul>
<li><code>Internal tree scale</code><ul>
<li><code>Display</code></li>
<li><code>Inverted scale</code>: <code>Yes</code></li>
</ul>
</li>
<li>Use the &ldquo;Fit to screen&rdquo; button <code>[]</code> to resize your tree in the viewer</li>
</ul>
<p><br>Note how the Tree scale - measured in millions of years - goes all the way back to 256, which is the minimum root age we applied as a constraint when using chronos to build the time tree from the molecular tree.</p>
<figure>
  <img src="../../../assets/images/day2/session3/itol_tree4.jpg" align="center" width=600/>
</figure>

<div class="admonition note">
<p class="admonition-title">If you did not manage to visualise the tree</p>
<p>You can find it on Gitpod at <code>/workspace/biodivinfo/data/Session3/Step1/tree_time_calibrated.jpg</code>. Or you can directly see it <a href="https://gitlab.com/evogenlab/teaching-repos/biodivinfo/-/blob/main/data/Session3/Step1/tree_time_calibrated.jpg">here</a>.</p>
</div>
<p><strong>Question: What are the main differences between the ultrametric time-calibrated tree shown above and our input molecular phylogeny copied again here below for comparison?</strong></p>
<details class="done">
<summary>Answer</summary>
<p>The branch lengths changed because the scale unit changed: this tree is expressed in time, while the previous one was expressed in substitutions/site. As a consequence, all the branches now have the same length, which is expected because all the species had the same span of time to evolve from their last common ancestor (<em>i.e.</em> the root of the tree).</p>
</details>
<figure>
  <img src="../../../assets/images/day2/session3/itol_tree2.jpg" align="center" width=600/>
</figure>

<p><strong>Questions:</strong></p>
<ul>
<li><strong>So what does &lsquo;ultrametric&rsquo; mean in the context of phylogenetic trees?</strong></li>
<li><strong>What might ultrametric trees be particularly useful for in terms of analyses?</strong></li>
</ul>
<details class="done">
<summary>Answer</summary>
<ul>
<li>An ultrametric tree is a rooted tree with edge lengths where all leaves are equidistant from the root. Here, it depicts times of divergence and as such can also be called a &lsquo;chronogram&rsquo;</li>
<li>Ultrametric trees can be useful to relate evolution events (appearance of a species, for example) with other events (drastic and brutal changes in climate, adaptation to new life conditions&hellip;)</li>
</ul>
</details>
<h2 id="building-a-tree-for-a-dynamic-gene-family">Building a tree for a dynamic gene family</h2>
<p>We will now follow the same steps that we did to compute a tree for a universal single-copy orthologous group in Session II, but this time selecting instead an orthologous group that has some species with more than one gene copy and some species missing an orthologue.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Remember that of the 14&lsquo;753 Coleoptera-level orthologous groups, only 1&lsquo;225 were showing single-copy orthologues in all 15 species. This means that the majority of orthologous groups show some level of dynamism in terms of gene gains and losses across this set of 15 beetle species.</p>
</div>
<p>Browsing OrthoDB I found an example orthologous group where some species have more than one copy and one species is lacking an orthologue: <a href="https://www.orthodb.org/?level=7041&amp;species=7041&amp;query=10261at7041">https://www.orthodb.org/?level=7041&amp;species=7041&amp;query=10261at7041</a>.</p>
<figure>
  <img src="../../../assets/images/day2/session3/orthodb_results1.jpg" align="center" width=500/>
</figure>

<p>It contains a total of 19 genes from 14 of the 15 Coleoptera species.</p>
<ul>
<li>10 species = single-copy (<em>i.e.</em> no gene duplications)</li>
<li>4 species = multi-copy (<em>i.e.</em> gene duplications)</li>
</ul>
<figure>
  <img src="../../../assets/images/day2/session3/orthodb_results2.jpg" align="center" width=500/>
</figure>

<p>Collect the input sequences:
<div class="highlight"><pre><span></span><code>curl -o 10261at7041.fasta &quot;https://data.orthodb.org/v11/fasta?id=10261at7041&amp;species=7041&quot;
</code></pre></div></p>
<p>Replace the colon &ldquo;_0:&rdquo; in the protein identifiers with an underscore &ldquo;_0_&ldquo;:
<div class="highlight"><pre><span></span><code>sed -i &#39;s/_0:/_0_/g&#39; 10261at7041.fasta
</code></pre></div></p>
<div class="admonition note">
<p class="admonition-title">If it failed, fetch the fasta file from the data files folder</p>
<div class="highlight"><pre><span></span><code>cp /workspace/biodivinfo/data/Session3/Step2/10261at7041.fasta .
</code></pre></div>
</div>
<p>Align the protein sequences - using MAFFT like before:
<div class="highlight"><pre><span></span><code>mafft --quiet --auto 10261at7041.fasta &gt; 10261at7041.aln.fasta
</code></pre></div></p>
<p>Filter/trim the alignment using TrimAl like before:
<div class="highlight"><pre><span></span><code>trimal -in 10261at7041.aln.fasta -out 10261at7041.aln.trm.fasta -automated1 -htmlout 10261at7041.aln.trm.html
</code></pre></div></p>
<p>Build the gene tree using RAxML like before:
<div class="highlight"><pre><span></span><code>raxmlHPC-SSE3 -s 10261at7041.aln.trm.fasta -f a -N 10 -p 12345 -x 12345 -m PROTGAMMAJTT -n 10261at7041
</code></pre></div></p>
<figure>
  <img src="../../../assets/images/day2/session3/raxml_run1.jpg" align="center" width=600/>
</figure>

<div class="admonition hint">
<p class="admonition-title">Hint</p>
<p>Note that RAxML is warning us that there are two pairs of sequences that are exactly identical and that these would normally be excluded from the analysis.</p>
</div>
<p>While RAxML is running you can download and view the trimmed alignment file (the HTML version) like we did before, to check out these identical sequences:</p>
<figure>
  <img src="../../../assets/images/day2/session3/raxml_msa1.jpg" align="center" width=250/>
</figure>

<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If you could not download it, you can view the alignment <a href="https://glcdn.githack.com/evogenlab/teaching-repos/biodivinfo/raw/main/data/Session3/Step2/10261at7041.aln.trm.html">here</a>.</p>
</div>
<figure>
  <img src="../../../assets/images/day2/session3/raxml_msa2.jpg" align="center" width=600/>
</figure>

<ul>
<li>Sequences <code>115357_0_00054b</code> and <code>115357_0_000a81</code> are exactly identical (pink)<ul>
<li>Two proteins from <em>Harmonia axyridis</em></li>
</ul>
</li>
<li>Sequences <code>7048_0_00018b</code> and <code>7048_0_0001a0</code> are exactly identical (orange)<ul>
<li>Two proteins from <em>Sitophilus oryzae</em></li>
</ul>
</li>
</ul>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>These proteins are not exactly identical along their entire lengths, but they are identical along the regions selected by TrimAl to use for phylogeny reconstruction. RAxML takes just one of any set of identical proteins to use for the phylogeny reconstruction and then adds back to the final tree any exact duplicates.</p>
</div>
<p><strong>Question: Why might there be some identical or near-identical protein sequences in the genomes of some species, <em>i.e.</em> what does it suggest about the evolution of these genes?</strong></p>
<details class="done">
<summary>Answer</summary>
<p>These genes probably underwent a duplication very recently. Only a short span of time passed since the duplication, which explains why they are onlly a few differences between the protein sequences.</p>
</details>
<div class="admonition note">
<p class="admonition-title">If these steps failed, you can fetch the results files from the data folder instead</p>
<div class="highlight"><pre><span></span><code>cp /workspace/biodivinfo/data/Session3/Step2/10261at7041.aln.fasta .
cp /workspace/biodivinfo/data/Session3/Step2/10261at7041.aln.trm.fasta .
cp /workspace/biodivinfo/data/Session3/Step2/10261at7041.aln.trm.html .
cp /workspace/biodivinfo/data/Session3/Step2/RAxML_info.10261at7041 .
cp /workspace/biodivinfo/data/Session3/Step2/RAxML_bootstrap.10261at7041 .
cp /workspace/biodivinfo/data/Session3/Step2/RAxML_bipartitions.10261at7041 .
cp /workspace/biodivinfo/data/Session3/Step2/RAxML_bestTree.10261at7041 .
</code></pre></div>
</div>
<p>When RAxML is finished you should have a message on the terminal that looks something like the following:</p>
<figure>
  <img src="../../../assets/images/day2/session3/raxml_run2.jpg" align="center" width=600/>
</figure>

<p>Now we open and view the resulting gene tree with bootstrapping support values (<code>RAxML_bipartitions.10261at7041</code>) using <a href="https://itol.embl.de/">iTOL</a> (as before).</p>
<div class="admonition hint">
<p class="admonition-title">Hint</p>
<p>Note the two pairs of &ldquo;identical&rdquo; proteins that RAxML warned us about earlier:</p>
<p><figure>
  <img src="../../../assets/images/day2/session3/itol_fam2_tree1.jpg" align="center" width=600/>
</figure></p>
</div>
<p>To make this gene tree more visually comparable to the species tree, let&rsquo;s start by properly rooting the tree according to where the known outgroup species (<code>7054_0</code> &amp; <code>224129_0</code>) branch off from the rest of the species (circled in pink):</p>
<figure>
  <img src="../../../assets/images/day2/session3/itol_fam2_tree2.jpg" align="center" width=600/>
</figure>

<p>We can use iTOL to interactively reroot the gene tree knowing the outgroup species. Click on the branch circled in pink shown above, hover over Tree structure, and then select <code>Re-root the tree here</code>.</p>
<figure>
  <img src="../../../assets/images/day2/session3/itol_fam2_settings1.jpg" align="center" width=250/>
</figure>

<p>In the re-rooted tree the outgroup species (blue) are now clearly placed to define the root of this phylogeny:</p>
<figure>
  <img src="../../../assets/images/day2/session3/itol_fam2_tree3.jpg" align="center" width=600/>
</figure>

<div class="admonition note">
<p class="admonition-title">If you did not manage to visualise the tree</p>
<p>You can find it on Gitpod at <code>/workspace/biodivinfo/data/Session3/Step2/tree_dynamic_family_rerooted.jpg</code>. Or you can directly see it <a href="https://gitlab.com/evogenlab/teaching-repos/biodivinfo/-/blob/main/data/Session3/Step2/tree_dynamic_family_rerooted.jpg">here</a>.</p>
</div>
<p><strong>Question: Can you manually (just by eye) identify where all speciation nodes, duplication nodes, and loss events have occurred based on this gene tree?</strong></p>
<div class="admonition hint">
<p class="admonition-title">Hint</p>
<p>When the identifiers start with the same numbers it means the genes are from the same species, <em>e.g.</em> species <code>7048_0</code> has three orthologues in this family.</p>
</div>
<details class="done">
<summary>Answer</summary>
<p>As mentioned previously, there was 2 duplication events (3 genes) within species <code>7048_0</code>. 1 duplication (2 genes) event happened within species <code>115357_0</code>. 1 speciation and 1 duplication events (4 genes) happened within species <code>7070_0</code> and <code>41895_0</code>. However, it is very hard to guess where the loss event happened with just a gene tree. It would help to have more information about the species tree.</p>
</details>
<h2 id="reconciling-the-gene-and-species-trees">Reconciling the gene and species trees</h2>
<p><img alt="" src="../../../assets/images/day2/session3/treerecs_logo.jpg" style="float: right;" width="250" /></p>
<p>Now we will try to see if an algorithm specifically designed to identify where all speciation nodes, duplication nodes, and loss events have occurred on a gene tree given a species tree agrees or disagrees with your manual assessments.</p>
<p>The main input requirements for <code>treerecs</code> are:</p>
<ul>
<li>A species phylogeny of all species: this can have more species than the gene tree but all species in the gene tree must be present in the species tree<ul>
<li>We have the <strong>ultrametric time-calibrated species tree</strong> we made earlier: <code>Coleoptera_TimeTree.tre</code></li>
<li>If you don&rsquo;t you can fetch it from the data folder:
<div class="highlight"><pre><span></span><code>cp /workspace/biodivinfo/data/Session3/Step1/Coleoptera_TimeTree.tre .
</code></pre></div></li>
</ul>
</li>
<li>A gene tree: this can have fewer species than the species tree (due to losses) and it can have more than one gene per species (due to duplications)<ul>
<li>We have the <strong>molecular gene family tree</strong> we made earlier: <code>RAxML_bipartitions.10261at7041</code></li>
<li>If you don&rsquo;t you can fetch it from the data folder:
<div class="highlight"><pre><span></span><code>cp /workspace/biodivinfo/data/Session3/Step2/RAxML_bipartitions.10261at7041 .
</code></pre></div></li>
</ul>
</li>
<li>
<p>A mapping file indicating which genes belong to which species: <code>treerecs</code> can try to do this by matching identifiers but it is advisable to provide a map</p>
<ul>
<li>This we <strong>made manually</strong>, you can fetch it from the data folder:
<div class="highlight"><pre><span></span><code>cp /workspace/biodivinfo/data/Session3/Step3/gene2species_map.txt .
</code></pre></div></li>
<li>This file simply lists the genes (green) and the species (blue) they belong to, so that <code>treerecs</code> knows which genes come from which species</li>
</ul>
<p><figure>
  <img src="../../../assets/images/day2/session3/mapping_file.jpg" align="center" width=250/>
</figure></p>
</li>
</ul>
<p>Start by checking out what <code>treerecs</code> needs as command-line options, input, and output names etc. in order to perform a gene-tree-species-tree reconciliation: <code>treerecs -h</code>.</p>
<figure>
  <img src="../../../assets/images/day2/session3/treerecs_usage.jpg" align="center" width=600/>
</figure>

<p>The main input that <code>treerecs</code> needs:</p>
<ul>
<li><code>-g</code>, <code>--genetree GENETREE_FILE</code></li>
<li><code>-s</code>, <code>--speciestree SPECIESTREE_FILE</code></li>
<li><code>-S</code>, <code>--smap SMAP_FILE</code></li>
<li>We will also ask it to: <code>-r</code>, <code>--reroot</code>: find the best root according to the reconciliation cost</li>
<li>We will ask it to produce both Newick and an image (.svg) output</li>
<li>And we will ask it to: <code>-f</code>, <code>--force</code>: force possible overwrite of existing files</li>
</ul>
<p>Putting them all together we have the following:
<div class="highlight"><pre><span></span><code>treerecs -g RAxML_bipartitions.10261at7041 -s Coleoptera_TimeTree.tre -S gene2species_map.txt -r -O newick:svg -f
</code></pre></div></p>
<figure>
  <img src="../../../assets/images/day2/session3/treerecs_run.jpg" align="center" width=600/>
</figure>

<p>This will create a folder called <code>treerecs_output</code>. The <code>.svg</code> output file can now be downloaded and viewed <em>e.g.</em> in a browser like Chrome. You can also click <a href="https://gitlab.com/evogenlab/teaching-repos/biodivinfo/-/blob/main/data/Session3/Step3/treerecs_output/RAxML_bipartitions.10261at7041_recs.svg">here</a> to view it directly.</p>
<figure>
  <img src="../../../assets/images/day2/session3/treerecs_tree1.jpg" align="center" width=250/>
</figure>

<div class="admonition note">
<p class="admonition-title">If treerecs failed you can fetch the files from the data folder</p>
<ul>
<li>The <code>.svg</code> file:
<div class="highlight"><pre><span></span><code>cp /workspace/biodivinfo/data/Session3/Step3/treerecs_output/RAxML_bipartitions.10261at7041_recs.svg .
</code></pre></div></li>
<li>The Newick file:
<div class="highlight"><pre><span></span><code>cp /workspace/biodivinfo/data/Session3/Step3/treerecs_output/RAxML_bipartitions.10261at7041_recs.nwk .
</code></pre></div></li>
</ul>
</div>
<p>Viewing the Newick file you will see that there is a header printed that provides the total &ldquo;cost&rdquo; and the number of predicted duplications and losses:</p>
<figure>
  <img src="../../../assets/images/day2/session3/treerecs_tree2.jpg" align="center" width=600/>
</figure>

<p><strong>Question: What is meant by the &ldquo;total cost&rdquo; of the reconciliation?</strong></p>
<details class="done">
<summary>Answer</summary>
<p>The total cost is the weighted sum of the cost of all the duplications and losses required to reconcile the two trees. Here, a duplication has a cost of 2 and a loss has a cost of 1 (5 * 2 + 4 * 1 = 14 = total cost).</p>
</details>
<p>The <code>.svg</code> image shows these results visually, by mapping the inferred evolutionary events (duplications  and losses ) inside the species phylogeny: showing a total of <strong>5 predicted duplications</strong> and <strong>4 predicted losses</strong>.</p>
<figure>
  <img src="../../../assets/images/day2/session3/treerecs_tree3.jpg" align="center" width=600/>
</figure>

<div class="admonition note">
<p class="admonition-title"><code>treerecs</code> symbols</p>
<ul>
<li>Duplication:
<img alt="" src="../../../assets/images/day2/session3/treerecs_dup.jpg" style="float: middle;" /></li>
<li>Loss:
<img alt="" src="../../../assets/images/day2/session3/treerecs_loss.jpg" style="float: middle;" /></li>
</ul>
</div>
<p>Let&rsquo;s now compare this with the input gene tree (<code>RAxML_bipartitions.10261at7041</code> - this should still be open in your iTOL browser), we can use iTOL to:</p>
<ul>
<li><code>Basic</code>: invert the gene tree</li>
<li><code>Basic</code>: ignore the branch lengths</li>
<li><code>Advanced</code>: resize to be taller and narrower</li>
</ul>
<figure>
  <img src="../../../assets/images/day2/session3/itol_fam2_settings2.jpg" align="center" width=600/>
</figure>

<div class="admonition note">
<p class="admonition-title">If you did not manage to visualise the tree</p>
<p>You can find it on Gitpod at <code>/workspace/biodivinfo/data/Session3/Step2/tree_dynamic_family_inverted.jpg</code>. Or you can directly see it <a href="https://gitlab.com/evogenlab/teaching-repos/biodivinfo/-/blob/main/data/Session3/Step3/tree_dynamic_family_inverted.jpg">here</a>.</p>
</div>
<p><strong>Question: Can you see the correspondence between the inferred events and the gene tree?</strong></p>
<details class="done">
<summary>Answer</summary>
<p>The correspondence appears quite clearly. However, it is easier to visualise with both trees side-by-side (see below).</p>
</details>
<figure>
  <img src="../../../assets/images/day2/session3/treerecs_tree4.jpg" align="center"/>
</figure>

<div class="admonition note">
<p class="admonition-title">High-resolution version</p>
<p>You can find a better version of the figure on Gitpod at <code>/workspace/biodivinfo/data/Session3/Step3/annotated_reconciliation.png</code>. Or you can directly see it <a href="https://gitlab.com/evogenlab/teaching-repos/biodivinfo/-/blob/main/data/Session3/Step3/annotated_reconciliation.png">here</a>.</p>
</div>
<p>The placement in the gene tree of the <em>Tenebrionoidea</em> (<em>Tribolium castaneum</em> &amp; <em>Tribolium madens</em>) as more early branching than the <em>Coccinelloidea</em> (<em>Coccinella septempunctata</em> &amp; <em>Hamonia axyridis</em>) is in conflict with the species tree. This means that reconciliation inferred <strong>1 duplication</strong> in the common ancestor of <em>Tenebrionoidea</em>-<em>Coccinelloidea</em> (pink) followed by <strong>3 losses</strong> to match the counts in the gene tree (green).</p>
<figure>
  <img src="../../../assets/images/day2/session3/treerecs_tree5.jpg" align="center"/>
</figure>

<p><strong>Question: Do you think this is a reasonable mapping of events - given that the node in the gene tree corresponding to <em>Tenebrionoidea</em>-<em>Coccinelloidea</em> obtained only 50% support?</strong></p>
<details class="done">
<summary>Answer</summary>
<p>This is probably not a reasonable mapping of event. Uncertainty in the gene tree is causing the reconciliation to over-complicate the number of events needed to resolve the two trees.</p>
</details>
<p>The good news is that <code>treerecs</code> can be asked to ignore weekly-supported nodes of a gene tree during the reconciliation process, setting a threshold <em>e.g.</em> of 55%:</p>
<ul>
<li>Achieved by adding the <code>-t 55</code> option</li>
<li>We also give the output a name <code>-o treerecs_output2</code> so it does not overwrite our first attempt at reconciliation using default parameters
<div class="highlight"><pre><span></span><code>treerecs -g RAxML_bipartitions.10261at7041 -s Coleoptera_TimeTree.tre -S gene2species_map.txt -r -O newick:svg -f -t 55 -o treerecs_output2
</code></pre></div></li>
</ul>
<div class="admonition note">
<p class="admonition-title">If treerecs failed you can fetch the files from the data folder</p>
<ul>
<li>The <code>.svg</code> file:
<div class="highlight"><pre><span></span><code>cp /workspace/biodivinfo/data/Session3/Step3/treerecs_output2/RAxML_bipartitions.10261at7041_recs.svg .
</code></pre></div></li>
<li>The Newick file:
<div class="highlight"><pre><span></span><code>cp /workspace/biodivinfo/data/Session3/Step3/treerecs_output2/RAxML_bipartitions.10261at7041_recs.nwk .
</code></pre></div>
When you download this new <code>.svg</code> make sure to change the name or it could overwrite the one you downloaded previously</li>
</ul>
<p><figure>
  <img src="../../../assets/images/day2/session3/treerecs_outputs.jpg" align="center" width=250/>
</figure></p>
</div>
<p>The result maps the inferred evolutionary events (duplications and losses) inside the species phylogeny: now <strong>4 duplications</strong> and <strong>1 loss</strong></p>
<ul>
<li>3 duplications appear as recent species-specific gene duplications</li>
<li>1 duplication appears in the ancestor of <em>Tribolium castaneum</em> &amp; <em>Tribolium madens</em></li>
</ul>
<figure>
  <img src="../../../assets/images/day2/session3/treerecs_tree6.jpg" align="center" width=500/>
</figure>

<p><strong>Questions:</strong></p>
<ul>
<li><strong>Do you think this is a more reasonable mapping of events?</strong></li>
<li><strong>Why is the <em>Tribolium</em> duplication considered as an ancestral event and not two independent duplications, one in each species of <em>Tribolium</em>?</strong></li>
</ul>
<details class="done">
<summary>Answer</summary>
<ul>
<li>This mapping of events is more reasonable because it decreases the number of events required to explain the phylogeny. As such, it respects the <a href="https://en.wikipedia.org/wiki/Maximum_parsimony_(phylogenetics)">principle of maximum parsimony</a></li>
<li>For the same reason than previously: it is more likely to see 1 ancestral duplication event followed by 1 speciation event rather than 1 speciation event followed by 2 ancestral duplication events</li>
</ul>
</details>
<figure>
  <img src="../../../assets/images/day2/session3/treerecs_tree7.jpg" align="center"/>
</figure>

<div class="admonition note">
<p class="admonition-title">High-resolution version</p>
<p>You can find a better version of the figure on Gitpod at <code>/workspace/biodivinfo/data/Session3/Step3/annotated_reconciliation2.png</code>. Or you can directly see it <a href="https://gitlab.com/evogenlab/teaching-repos/biodivinfo/-/blob/main/data/Session3/Step3/annotated_reconciliation2.png">here</a>.</p>
</div>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright &copy; 2021-2025 SIB Swiss institute of Bioinformatics - <a href="#__consent">Change cookie settings</a>
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "../../..", "features": ["announce.dismiss", "content.code.copy"], "search": "../../../assets/javascripts/workers/search.d50fe291.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": {"provider": "mike"}}</script>
    
    
      <script src="../../../assets/javascripts/bundle.13a4f30d.min.js"></script>
      
    
  </body>
</html>