
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="../2_software/">
      
      
        <link rel="next" href="../4_group_function/">
      
      
      <link rel="icon" href="../../../assets/images/general/SIB_logo.svg">
      <meta name="generator" content="mkdocs-1.5.3, mkdocs-material-9.4.7">
    
    
      
        <title>Finding information on gene function at small scale - Biodiversity bioinformatics</title>
      
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/main.4b4a2bd9.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../../stylesheets/extra.css">
    
    <script>__md_scope=new URL("../../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    <body dir="ltr">
  
    
    
      <script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#aim" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <div data-md-color-scheme="default" data-md-component="outdated" hidden>
        
          <aside class="md-banner md-banner--warning">
            <div class="md-banner__inner md-grid md-typeset">
              
  You're not viewing the latest version.
  <a href="../../../.."> 
    <strong>Click here to go to latest.</strong>
  </a>

            </div>
            <script>var el=document.querySelector("[data-md-component=outdated]"),outdated=__md_get("__outdated",sessionStorage);!0===outdated&&el&&(el.hidden=!1)</script>
          </aside>
        
      </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../../.." title="Biodiversity bioinformatics" class="md-header__button md-logo" aria-label="Biodiversity bioinformatics" data-md-component="logo">
      
  <img src="../../../assets/images/general/SIB_logo.svg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Biodiversity bioinformatics
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Finding information on gene function at small scale
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/sib-swiss/biodiversity-bioinformatics" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.4.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    sib-swiss/biodiversity-bioinformatics
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../../.." title="Biodiversity bioinformatics" class="md-nav__button md-logo" aria-label="Biodiversity bioinformatics" data-md-component="logo">
      
  <img src="../../../assets/images/general/SIB_logo.svg" alt="logo">

    </a>
    Biodiversity bioinformatics
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/sib-swiss/biodiversity-bioinformatics" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.4.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    sib-swiss/biodiversity-bioinformatics
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
    <li class="md-nav__item">
      <a href="../../.." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Home
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
    <li class="md-nav__item">
      <a href="../../../precourse/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Precourse preparations and Gitpod information
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
    <li class="md-nav__item">
      <a href="../../../course_schedule/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Course schedule
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" >
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Day 1 - Introduction to molecular evolution & phylogenetics
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            Day 1 - Introduction to molecular evolution & phylogenetics
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../../day1/1_general/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    General information and course material
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../../day1/2_exercices/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Links to the OMA Academy modules
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" >
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Day 2 - Orthology-focused arthropod comparative genomics
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            Day 2 - Orthology-focused arthropod comparative genomics
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../../day2/1_general/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    General information and course material
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../../day2/2_software/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Software and resources
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../../day2/3_busco/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    BUSCO - the what, why, and how!
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../../day2/4_homology/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Homology, Orthology, & Paralogy
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../../day2/5_reconciliation/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Gene-Tree-Species-Tree Reconciliation
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../../day2/6_reconstruction/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Gene Family Ancestral State Reconstruction
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6" checked>
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Day 3 - Duplication of genes and genomes, expression
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            Day 3 - Duplication of genes and genomes, expression
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../1_general/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    General information and course material
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../2_software/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Software and R packages
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    Finding information on gene function at small scale
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    Finding information on gene function at small scale
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#aim" class="md-nav__link">
    Aim
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#learning-outcomes" class="md-nav__link">
    Learning outcomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" class="md-nav__link">
    Introduction
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#check-about-ncdn-in-uniprot" class="md-nav__link">
    Check about NCDN in Uniprot
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#learn-about-ncdns-gene-ontology-annotation" class="md-nav__link">
    Learn about NCDN's Gene Ontology annotation
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#look-at-expression-of-ncdn-for-several-species-in-bgee" class="md-nav__link">
    Look at expression of NCDN for several species in Bgee
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../4_group_function/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Finding information about function of a group of genes
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../5_duplicated_function/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Function of duplicated genes
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#aim" class="md-nav__link">
    Aim
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#learning-outcomes" class="md-nav__link">
    Learning outcomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" class="md-nav__link">
    Introduction
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#check-about-ncdn-in-uniprot" class="md-nav__link">
    Check about NCDN in Uniprot
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#learn-about-ncdns-gene-ontology-annotation" class="md-nav__link">
    Learn about NCDN's Gene Ontology annotation
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#look-at-expression-of-ncdn-for-several-species-in-bgee" class="md-nav__link">
    Look at expression of NCDN for several species in Bgee
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
  


  <h1>Finding information on gene function at small scale</h1>

<h2 id="aim">Aim</h2>
<p>Use the gene <code>NCDN</code> as a case example and investigate how to know about a gene&rsquo;s function.</p>
<h2 id="learning-outcomes">Learning outcomes</h2>
<ul>
<li>What can we learn about <code>NCDN</code> function determination in Uniprot?</li>
<li>What does <code>NCDN</code> Gene Onthology (GO) annotation tells us about its function?</li>
<li>What can we deduce from <code>NCDN</code> gene expression in several species?</li>
</ul>
<h2 id="introduction">Introduction</h2>
<p>We will start the hands-on part of this course by focusing on a single gene as a case study. In this case, we will use the gene <code>NCDN</code> (neurochondrin).</p>
<h2 id="check-about-ncdn-in-uniprot">Check about <code>NCDN</code> in Uniprot</h2>
<p>Go to <a href="https://www.uniprot.org/">Uniprot</a> and find the <strong>human</strong> gene <code>NCDN</code> page (<code>NCDN_HUMAN</code>). There is a lot of information within a Uniprot&rsquo;s protein page; let&rsquo;s first focus on the function description.</p>
<div class="admonition hint">
<p class="admonition-title">Hint</p>
<p>Leave all the different species gene pages open in tabs. We will come back to them!</p>
</div>
<p><strong>Questions:</strong></p>
<ul>
<li><strong>What can we learn from the description of the human <code>NCDN</code>&rsquo;s function?</strong></li>
<li><strong>How was this information annotated?</strong></li>
</ul>
<details class="done">
<summary>Answer</summary>
</details>
<p>Now go to the <strong>mouse</strong> gene <code>NCDN</code> Uniprot page (<code>NCDN_MOUSE</code>).</p>
<p><strong>Questions:</strong></p>
<ul>
<li><strong>What can we learn from the description of the mouse <code>NCDN</code>&rsquo;s function?</strong></li>
<li><strong>How was this information annotated?</strong></li>
</ul>
<details class="done">
<summary>Answer</summary>
</details>
<p>Try it again with the <strong>bovin</strong> gene (<code>NCDN_BOVIN</code>), the <strong>drosophila</strong> gene page (<code>NCDN_DROME</code>) and the <strong>xenopus</strong> gene page (<code>NCDN_XENLA</code>).</p>
<p><strong>Questions:</strong></p>
<ul>
<li><strong>What can we learn about Uniprot&rsquo;s gene function assessment?</strong></li>
<li><strong>In which species is most of the research on <code>NCDN</code> done?</strong></li>
</ul>
<details class="done">
<summary>Answer</summary>
</details>
<p>Two other ways to shed light into a gene&rsquo;s function are checking its subcellular localization and its sequence features (domains, variantsâ€¦). We can find information on both of these aspects in Uniprot. First, focus on the subcellular location. There is one section dedicated to this in all Uniprot gene pages. Explore all the ones we opened before.</p>
<p><strong>Questions:</strong></p>
<ul>
<li><strong>Is <code>NCDN</code> subcellular location informative on its function?</strong></li>
<li><strong>From where was each information retrieved?</strong></li>
</ul>
<details class="done">
<summary>Answer</summary>
</details>
<p>Now try the same with the sequence features. There is a new <code>Feature viewer</code> in Uniprot gene pages. Explore it for all the species above.</p>
<p><strong>Questions:</strong></p>
<ul>
<li><strong>Is the <code>NCDN</code> sequence informative?</strong></li>
<li><strong>Does it mean that retrieving information from a protein sequence is meaningless?</strong></li>
</ul>
<details class="done">
<summary>Answer</summary>
</details>
<h2 id="learn-about-ncdns-gene-ontology-annotation">Learn about <code>NCDN</code>&rsquo;s Gene Ontology annotation</h2>
<p>Now, we will focus on <code>NCDN</code>&rsquo;s Gene Ontology (GO) annotation. There are many tools that use GO annotation in different ways. In this practical, we will use some of the most established ones.</p>
<p>First, still in the Uniprot gene pages. Go to the GO annotation section on each page. Compare the GO terms in which <code>NCDN</code> is annotated in each species and the sources of such annotations.</p>
<p><strong>Question: What can we learn on GO annotations in non-model species?</strong></p>
<p>Go to the <a href="https://amigo.geneontology.org/amigo/landing">AmiGO 2 website</a>, in the top menu, select <code>Search</code> and then <code>Genes and gene products</code>. Search for our <code>NCDN</code> gene.</p>
<p><strong>Questions:</strong></p>
<ul>
<li><strong>For which species do we have information? Why?</strong></li>
<li><strong>How can we get GO annotations for other species?</strong></li>
</ul>
<details class="done">
<summary>Answer</summary>
</details>
<p>Now you can go to each species AmiGO 2 <code>NCDN</code> pages. Explore them a bit.</p>
<p><strong>Question: From where do they retrieve the evidence for the GO annotations?</strong></p>
<details class="done">
<summary>Answer</summary>
</details>
<h2 id="look-at-expression-of-ncdn-for-several-species-in-bgee">Look at expression of <code>NCDN</code> for several species in Bgee</h2>
<p>We are now going to utilize the Bgee database in order to retrieve expression related information on our target gene (<code>NCDN</code>). You can access the webpage of Bgee in your browser by following <a href="https://www.bgee.org/">this link</a>.</p>
<p>You will arrive at the Bgee homepage where you can access the various tools of Bgee in the top toolbar, directly search for a gene in the middle search box or download expression data for various animal species by clicking on the correct species picture in the bottom of the page. For this step, we will be interested in continuing to explore the function of <code>NCDN</code> orthologs, now focusing on their expression. To do this, you will just need to search for a gene name (<code>NCDN</code>), a UniProt or an Ensembl identifier.</p>
<figure>
  <img src="../../../assets/images/day3/session1/gene_search.gif" align="center"/>
</figure>

<p>Doing this, you will be directed to the gene search result page where you can see that there are 49 genes in Bgee for the gene name <code>NCDN</code>.</p>
<p><strong>Question: Why are there multiple Ensembl identifiers for one gene name (<code>NCDN</code>)?</strong></p>
<details class="done">
<summary>Answer</summary>
</details>
<p>In this page, you will notably find a column for <code>Gene identifier</code> that represents the unique Ensembl identifiers for each species or the species scientific and common name in the column <code>Organism</code>. To continue, please select a species (for example <strong>mouse</strong>) and click on their Ensembl ID (<a href="https://www.bgee.org/gene/ENSMUSG00000028833/">ENSMUSG00000028833</a>). This will lead you to the mouse specific gene page for <code>NCDN</code>; don&rsquo;t worry you will still be able to see other orthologs of <code>NCDN</code>.</p>
<figure>
  <img src="../../../assets/images/day3/session1/gene_search.jpg" align="center"/>
</figure>

<p>On the left, you have the headers that allow you to jump easily to different sections of the page such as Expression or Orthologs. Since in this exercise we want to know where <code>NCDN</code> is expressed, we will primarily focus on the expression section for now.</p>
<figure>
  <img src="../../../assets/images/day3/session1/expression_celltype.jpg" align="center">
</figure>

<p>In this expression tab, you will find all anatomical entities where Bgee has evidence that this gene is expressed. You will notably find the expression score that denotes how strongly the gene was expressed in that tissue (from 0 to 100). If you want to verify the given information, you can also access the data used to determine expression in the link to source data column or see which type of data was used in source (such as R for bulk RNA-seq or A for Affymetrix microarray).</p>
<p><strong>Question: Looking at the top entries in the expression table, in which organ(s) do you think this gene is generally expressed?</strong></p>
<details class="done">
<summary>Answer</summary>
</details>
<p>We can also investigate if this gene is expressed at different stages of development. To do so, you can unselect <code>Anat. entity and cell types</code> and tick the <code>Dev. stage</code> box on top of the expression tab and update the results.</p>
<figure>
  <img src="../../../assets/images/day3/session1/expression_devstage.jpg" align="center">
</figure>

<p>Here, we see that this gene seems to be expressed in late adults as well as the embryonic stage (for example, <code>Theiler stage 04</code>). If you are unsure what a developmental stage term means, clicking on the identifier (<em>e.g.</em> <code>MmusDv:0000008</code>) will take you to the description of this term in the developmental ontology. Interestingly, in this case, looking at the <code>Reported absence of expression</code> section (below in the same page), we can see that there are no developmental stages where the gene is not expressed at all, even though its expression is much lower outside the brain.</p>
<figure>
  <img src="../../../assets/images/day3/session1/gene_noexpression.jpg" align="center">
</figure>

<p><strong>Questions:</strong></p>
<ul>
<li><strong>Go back to the <code>Expression</code> section and the <code>Anat. entity and cell types</code> filter. Where is <code>NCDN</code> the least expressed?</strong></li>
<li><strong>Play around by combining the filters for sex, strain, developmental stage or anatomical entities, which insights can you gather?</strong></li>
</ul>
<details class="done">
<summary>Answer</summary>
</details>
<p>By clicking on the <code>Orthologs</code> section in the menu on the left or scrolling down to it, you can find the different orthologs for NCDN which are found in Bgee. This information is retrieved from the OMA SPARQL endpoint that you used on the first day. Orthologs for NCDN are found at 11 different taxon levels. You can see the Ensembl identifiers of orthologous genes at specific taxon levels by clicking the <code>See details</code> arrow.</p>
<figure>
  <img src="../../../assets/images/day3/session1/gene_orthologs.jpg" align="center">
</figure>

<p><strong>Question: Which is the higher taxon in which you can find human orthologs of mouse <code>NCDN</code>?</strong></p>
<details class="done">
<summary>Answer</summary>
</details>
<p>Unfold the orthologs (<code>See details</code>) to find the human <code>NCDN</code>, and open its page in a new tab.</p>
<p><strong>Question: Does the function of <code>NCDN</code> seem to have been conserved between mouse and human? What about more distant species?</strong></p>
<details class="done">
<summary>Answer</summary>
</details>
<p>Now we are going to compare the expression of the different orthologs of <code>NCDN</code> for Xenopus, Human, mouse, rat, guinea pig and macaque.</p>
<p>First, gather the Ensembl identifiers for those genes by looking at the orthologs tab!</p>
<ul>
<li>Rat (<code>ENSRNOG00000011751</code>)</li>
<li>Xenopus (<code>ENSXETG00000009669</code>)</li>
<li>Human (<code>ENSG00000020129</code>)</li>
<li>Mouse (<code>ENSMUSG00000028833</code>)</li>
<li>Guinea pig (<code>ENSCPOG00000023962</code>)</li>
<li>Macaque (<code>ENSMMUG00000019812</code>).</li>
</ul>
<p>We will learn to use Bgee expression comparison tool in order to gain novel insights on <code>NCDN</code> anatomical expression. To do this, click on the Analysis section of the top toolbar of Bgee and then on Expression comparison.</p>
<figure>
  <img src="../../../assets/images/day3/session1/expression_comparison_tool.gif" align="center">
</figure>

<p>Fill the orthologs identifiers in the gene list, with one gene per line and click the search button to launch Bgee&rsquo;s expression comparison tool.</p>
<figure>
  <img src="../../../assets/images/day3/session1/expression_comparison_list.jpg" align="center" width=500>
</figure>

<p>You will obtain the following results:</p>
<figure>
  <img src="../../../assets/images/day3/session1/expression_comparison_results.jpg" align="center">
</figure>

<p><strong>Question: Looking at the max expression scores, do the results seem consistent with the anatomical localization of <code>NCDN</code> expression when looking at in individual species?</strong></p>
<details class="done">
<summary>Answer</summary>
</details>
<p>The conservation score represents the ratio between genes being called present in the tissue and the total number of genes. In our case since we only used orthologs, this value gives us the ratio between species which have <code>NCDN</code> expressed in that tissue divided by the total number of species for which we have data in that condition (maximum 6).</p>
<p><strong>Question: With the help of the conservation score, can you see for which species and tissue there is a qualitative difference in the expression of <code>NCDN</code>?</strong></p>
<details class="done">
<summary>Answer</summary>
</details>





                
              </article>
            </div>
          
          
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      <a href="#__consent">Change cookies settings</a>
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "../../..", "features": ["announce.dismiss"], "search": "../../../assets/javascripts/workers/search.f886a092.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": {"provider": "mike"}}</script>
    
    
      <script src="../../../assets/javascripts/bundle.aecac24b.min.js"></script>
      
    
  </body>
</html>