## General objectives

Interpret phylogenetic trees and alignments; general understanding of how trees are inferred; knowing the definition of orthology, paralogy and their subtypes; understanding the difference between pairwise and groupwise comparisons; ability to perform orthology inference through tree overlap method; ability to retrieve orthology information from the OMA database; ability to infer orthologs using the OMA standalone pipeline.

## Course material

### Lecture: Overview, objectives, motivation, concept of orthologs, HOGs

[:fontawesome-solid-file-pdf: Download the presentation](../../assets/pdf/day1/lecture_Natasha_GLOVER.pdf){: .md-button }

### Module 1: Genes, groups, and genomes in the OMA Browser

[:fontawesome-solid-file-pdf: Download the presentation](../../assets/pdf/day1/module1_Natasha_GLOVER.pdf){: .md-button }

### Module 2: Fast placement with OMAmer

[:fontawesome-solid-file-pdf: Download the presentation](../../assets/pdf/day1/module2_Yannis_NEVERS.pdf){: .md-button }

### Module 3: FastOMA

[:fontawesome-solid-file-pdf: Download the presentation](../../assets/pdf/day1/module3_Sina_MAJIDIAN.pdf){: .md-button }

### Module 4: Building Species Trees

[:fontawesome-solid-file-pdf: Download the presentation](../../assets/pdf/day1/module4_Christophe_DESSIMOZ.pdf){: .md-button }

## Course gitlab repository

The gitlab repository to run all the practicals is available at [https://github.com/DessimozLab/SIBBiodiversityBioinformatics2024](https://github.com/DessimozLab/SIBBiodiversityBioinformatics2024).

!!! note
	Remember to open a Gitpod workspace before starting the practicals. Click [here](../../precourse.md#software) for more information on Gitpod workspace.

## Working directories

You can find the working directories of Modules 2, 3, and 4 in the following list:

* [Module 2 working directory](https://github.com/DessimozLab/SIBBiodiversityBioinformatics2024/tree/main/Module2_OMAmer/working_dir)
* [Module 3 working directory](https://github.com/DessimozLab/SIBBiodiversityBioinformatics2024/tree/main/Module3_FastOMA/working_dir/in_folder)
* [Module 4 working directory](https://github.com/DessimozLab/SIBBiodiversityBioinformatics2024/tree/main/Module4_SpeciesTrees/working_dir)

## Exercises ouputs

You can find the expected outputs of Modules 2, 3, and 4 in the following list:

* [Module 2 expected outputs](https://github.com/DessimozLab/SIBBiodiversityBioinformatics2024/tree/main/Module2_OMAmer/expected_output)
* [Module 3 expected outputs](https://github.com/DessimozLab/SIBBiodiversityBioinformatics2024/tree/main/Module3_FastOMA/expected_output)
* [Module 4 expected outputs](https://github.com/DessimozLab/SIBBiodiversityBioinformatics2024/tree/main/Module4_SpeciesTrees/expected_output)
